### PR TITLE
Add the needed StringInfo methods to Linux implementation

### PR DESCRIPTION
**Pull request by tarekgh**

### CODE REVIEW

**Comment by tarekgh** [comment]
@jkotas @weshaggard @danmosemsft

File: src/mscorlib/corefx/System/Globalization/StringInfo.cs [hunk 1]
```diff
@@ -119,7 +119,61 @@ public int LengthInTextElements
...
+                }
+            }
+            return (SubstringByTextElements(startingTextElement, Indexes.Length - startingTextElement));
+        }
+
+        public string SubstringByTextElements(int startingTextElement, int lengthInTextElements) 
+        {
+            if (startingTextElement < 0)
+            {
+                throw new ArgumentOutOfRangeException("startingTextElement", SR.ArgumentOutOfRange_NeedPosNum);
```
Comment (danmosemsft) [inline src/mscorlib/corefx/System/Globalization/StringInfo.cs]: >"startingTextElement" [](start = 54, length = 21)

can you replace all your quoted strings with eg nameof(startingTextElement) ?

**Comment by danmosemsft** [comment]
:shipit:

File: src/mscorlib/corefx/System/Globalization/StringInfo.cs (hunk 1 above)
Comment (jkotas) [inline src/mscorlib/corefx/System/Globalization/StringInfo.cs]: We have not done the nameof reformatting in CoreCLR and CoreRT yet; and our coding guidelines say to prefer consistency with the existing style...

**Pull request merged**